crates/viewer/re_renderer/src/video/mod.rs

```diff
@@ -66,25 +77,30 @@ impl Video {
         data: &[u8],
         media_type: Option<&str>,
     ) -> Result<Self, VideoError> {
-        let data = re_video::VideoData::load_from_bytes(data, media_type)?;
-        let decoder = decoder::VideoDecoder::new(render_context, data)?;
+        let data = Arc::new(re_video::VideoData::load_from_bytes(data, media_type)?);
+        let decoder = Mutex::new(decoder::VideoDecoder::new(render_context, data.clone())?);
 
-        Ok(Self { decoder })
+        Ok(Self { data, decoder })
     }
 
     /// Duration of the video in milliseconds.
     pub fn duration_ms(&self) -> f64 {
-        self.decoder.duration_ms()
+        self.data.duration.as_f64()
```

Comment:
Nit: I have seen so many time unit bugs in my days that I always go for explicitness

```suggestion
        self.data.duration_ms.as_f64()
```

Reply:
`duration` is of type `TimeMs`. We should probably just forward that!